crates/notedeck_columns/src/ui/settings.rs

```diff
@@ -1,22 +1,93 @@
-use egui::{vec2, Button, Color32, ComboBox, Frame, Margin, RichText, ThemePreference};
-use notedeck::{tr, Images, LanguageIdentifier, Localization, NotedeckTextStyle, ThemeHandler};
+use egui::{vec2, Button, Color32, ComboBox, Frame, Margin, RichText, ScrollArea, ThemePreference};
+use notedeck::{
+    tr, Images, LanguageIdentifier, Localization, NotedeckTextStyle, Settings, SettingsHandler,
+};
 use notedeck_ui::NoteOptions;
 use strum::Display;
 
 use crate::{nav::RouterAction, Damus, Route};
 
+const THEME_LIGHT: &str = "Light";
+const THEME_DARK: &str = "Dark";
+
+const MIN_ZOOM: f32 = 0.5;
+const MAX_ZOOM: f32 = 3.0;
+const ZOOM_STEP: f32 = 0.1;
+const RESET_ZOOM: f32 = 1.0;
+
 #[derive(Clone, Copy, PartialEq, Eq, Display)]
-pub enum ShowNoteClientOptions {
+pub enum ShowSourceClientOption {
     Hide,
     Top,
     Bottom,
 }
 
+impl Into<String> for ShowSourceClientOption {
+    fn into(self) -> String {
+        match self {
+            Self::Hide => "hide".to_string(),
+            Self::Top => "top".to_string(),
+            Self::Bottom => "bottom".to_string(),
+        }
+    }
+}
+
+impl From<NoteOptions> for ShowSourceClientOption {
+    fn from(note_options: NoteOptions) -> Self {
+        if note_options.contains(NoteOptions::ShowNoteClientTop) {
+            ShowSourceClientOption::Top
+        } else if note_options.contains(NoteOptions::ShowNoteClientBottom) {
+            ShowSourceClientOption::Bottom
+        } else {
+            ShowSourceClientOption::Hide
+        }
+    }
+}
+
+impl From<String> for ShowSourceClientOption {
+    fn from(s: String) -> Self {
+        match s.to_lowercase().as_str() {
+            "hide" => Self::Hide,
+            "top" => Self::Top,
+            "bottom" => Self::Bottom,
+            _ => Self::Hide, // default fallback
+        }
+    }
+}
+
+impl ShowSourceClientOption {
+    pub fn set_note_options(self, note_options: &mut NoteOptions) {
+        match self {
+            Self::Hide => {
+                note_options.set(NoteOptions::ShowNoteClientTop, false);
+                note_options.set(NoteOptions::ShowNoteClientBottom, false);
+            }
+            Self::Bottom => {
+                note_options.set(NoteOptions::ShowNoteClientTop, false);
+                note_options.set(NoteOptions::ShowNoteClientBottom, true);
+            }
+            Self::Top => {
+                note_options.set(NoteOptions::ShowNoteClientTop, true);
+                note_options.set(NoteOptions::ShowNoteClientBottom, false);
+            }
+        }
+    }
+
+    fn label<'a>(&self, i18n: &'a mut Localization) -> String {
+        match self {
+            Self::Hide => tr!(i18n, "Hide", "Hide client source label"),
+            Self::Top => tr!(i18n, "Top", "Client source at top"),
+            Self::Bottom => tr!(i18n, "Bottom", "Client source at bottom"),
```

Comment:
I'd suggest having more descriptive localization comments to make it easier for translators to be able to translate. These are also the comments I have on my unmerged branch that I've exported to Crowdin for translation already.

```suggestion
            Self::Hide => tr!(i18n, "Hide", "Option in settings section to hide the source client label in note display"),
            Self::Top => tr!(i18n, "Top", "Option in settings section to show the source client label at the top of the note"),
            Self::Bottom => tr!(i18n, "Bottom", "Option in settings section to show the source client label at the bottom of the note"),
```